File: reamenus.py
from pymenuset import *


def build(menuset=None):
	if menuset is None:
		menuset = MenuSet()

	# Main menus
	menuset.menus['Main track'].items = menuset.menus['Track control panel context'].items[:]
	menuset.menus['Main track'].title = '&Track'

	menuset.menus['Main item'].items = menuset.menus['Media item context'].items[:]
	menuset.menus['Main item'].title = '&Item'

	menuset.menus['Main insert'].items = menuset.menus['Ruler/arrange context'].items[:]
	menuset.menus['Main insert'].title = 'Time&line'

	menuset.menus['Main extensions'].title = '&Memory'

	# MIDI main navigate
	menuset.menus['MIDI main navigate'].items = menuset.menus['MIDI piano roll context'].items[:]
	menuset.menus['MIDI main navigate'].title = '&Notes'

	for item in menuset.menus['MIDI main navigate'].items:
		if type(item) is Submenu and 'Velocity list' in item.name:
			vel_split = menuset.menus['MIDI main navigate'].items.index(item)
			break

	del menuset.menus['MIDI main navigate'].items[vel_split]

	# Empty TCP context
	menuset.menus['Empty TCP context'].items = menuset.menus['Track control panel context'].items[:]

	# Envelope context
	for item in menuset.menus['Envelope context'].items:
		if type(item) is Separator:
			env_index = menuset.menus['Envelope context'].items.index(item)
			env_split = menuset.menus['Envelope context'].items[env_index:]
			break

	for item in menuset.menus['Envelope point context'].items:
		if type(item) is Separator:
			pnt_index = menuset.menus['Envelope point context'].items.index(item)
			pnt_split = menuset.menus['Envelope point context'].items[:pnt_index]
			break

	menuset.menus['Envelope point context'].items = pnt_split + env_split

	# MIDI main menu context
	for item in menuset.menus['MIDI main menu context'].items:
		if type(item) is not Submenu:
			continue

		if 'File' in item.name:
			item.items = menuset.menus['MIDI main file'].items[:]
		elif 'Edit' in item.name:
			item.items = menuset.menus['MIDI main edit'].items[:]
		elif 'Notes' in item.name:
			item.items = menuset.menus['MIDI main navigate'].items[:]
		elif 'Options' in item.name:
			item.items = menuset.menus['MIDI main options'].items[:]
		elif 'View' in item.name:
			item.items = menuset.menus['MIDI main view'].items[:]
		elif 'Actions' in item.name:
			item.items = menuset.menus['MIDI main actions'].items[:]

	# wrap it up
	menuset.strip_toolbars().style().write()


if __name__ == '__main__':
	build()

File: reamenus.js
'use strict';
require("babel/register");

// import

const _ = require('lodash');
const fs = require('fs-extra');
const async = require('async');

const MenuSet = require('./models/MenuSet');

// build

function build(menuset) {

  // main menus

  menuset.menus['Main track'].items = menuset.menus['Track control panel context'].items.slice();
  menuset.menus['Main track'].title = '&Track';

  menuset.menus['Main item'].items = menuset.menus['Media item context'].items.slice();
  menuset.menus['Main item'].title = '&Item';

  menuset.menus['Main insert'].items = menuset.menus['Ruler/arrange context'].items.slice();
  menuset.menus['Main insert'].title = 'Time&line';

  menuset.menus['Main extensions'].title = '&Memory';

  // midi menus

  menuset.menus['MIDI main navigate'].items = menuset.menus['MIDI piano roll context'].items.slice();
  menuset.menus['MIDI main navigate'].title = '&Notes';

  let split = undefined;

  menuset.menus['MIDI main navigate'].items.some((item, i) => {
    if (_.includes(item.name, 'Velocity list')) {
      menuset.menus['MIDI main navigate'].items.splice(i, 1);
      return true;
    }
  });

  menuset.menus['MIDI main menu context'].items.forEach((item) => {
    if (!item.items) {
      return;
    }

    switch (item.name.replace(/&/g, '').trim()) {
      case 'File':
        return item.items = menuset.menus['MIDI main file'].items.slice();
      case 'Edit':
        return item.items = menuset.menus['MIDI main edit'].items.slice();
      case 'Notes':
        return item.items = menuset.menus['MIDI main navigate'].items.slice();
      case 'Options':
        return item.items = menuset.menus['MIDI main options'].items.slice();
      case 'View':
        return item.items = menuset.menus['MIDI main view'].items.slice();
      case 'Contents':
        return item.items = menuset.menus['MIDI main contents'].items.slice();
      case 'Actions':
        return item.items = menuset.menus['MIDI main actions'].items.slice();
    }
  });

  // envelope menus

  menuset.menus['Envelope context'].items.some((envItem, e) => {
    if (envItem.constructor.name != 'Separator') {
      return;
    }

    return menuset.menus['Envelope point context'].items.some((pntItem, p) => {
      if (pntItem.constructor.name != 'Separator') {
        return;
      }

      return menuset.menus['Envelope point context'].items = menuset.menus['Envelope point context'].items.slice(0, e).concat(menuset.menus['Envelope point context'].items.slice(p));
    });
  });

  // track menus

  menuset.menus['Empty TCP context'].items = menuset.menus['Track control panel context'].items.slice();

  // toolbars

  menuset.menus = _.omit(menuset.menus, (menu, name) => {
    return _.includes(name.toLowerCase(), 'toolbar');
  });

  // wrap up

  menuset.style().write();
}

// run

fs.readdir('./input', (err, files) => {
  async.each(files, (menuset) => {
    build(new MenuSet('./input/' + menuset, './output/' + menuset));
  });
});
